targets/linux/distro_handler.go

```diff
@@ -90,30 +93,48 @@ func HandleContainer(c DistroConfig) gwclient.BuildFunc {
 				return nil, nil, err
 			}
 
-			pg := dalec.ProgressGroup(spec.Name)
-			pc := dalec.Platform(platform)
+			var opts []llb.ConstraintsOpt
+			opts = append(opts, dalec.ProgressGroup(spec.Name))
+			opts = append(opts, dalec.Platform(platform))
 
-			worker, err := c.Worker(sOpt, pg, pc)
+			worker, err := c.Worker(sOpt, opts...)
 			if err != nil {
 				return nil, nil, err
 			}
 
-			deb, err := c.BuildPkg(ctx, client, worker, sOpt, spec, targetKey, pg, pc)
+			// We need to get the build context from the client.
+			inputs, err := client.Inputs(ctx)
 			if err != nil {
-				return nil, nil, err
+				return nil, nil, fmt.Errorf("error getting llb state from client with provided build context: %w", err)
+			}
+
+			// If we have a build context named after the target, then we should use that pre-built
+			// package when building the container.
+			var pkgSt llb.State
+			if input, ok := inputs[targetKey]; ok {
```

Comment:
`sOpt.GetContext`
This makes sure it obeys include/exclude filters.
Note this can return a `nil` state, which means no such context was set.

Reply:
I think `targetKey` is not the best name to use here.
Maybe something like `pkg-dir`?
Maybe it should include the `targetKey` in there, but then it gets more complicated to use it since the client needs to derive the target to setup the build.

Or perhaps either `targetKey+"-pkg-dir"` OR `pkg-dir`, with dalec preferring the former if it is set.

Another route would be to set it to the name of the build target (e.g. azlinux3/rpm) that produced the output.
Each target (e.g. `azlinux3`) can specify the name they are expecting to lookup from the inputs.

Anyway, naming is hard, just some suggestions here.